helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -331,7 +331,7 @@ private void reportTopStateHandoffFailIfNecessary(ResourceControllerDataProvider
       missingTopStateMap.get(resourceName).put(partitionName, record);
       // Since top state handoff has not completed yet we can't log helix top state latency but can log since how long
       // top state is missing.
-      LogUtil.logDebug(LOG, _eventId, String.format(
+      LogUtil.logInfo(LOG, _eventId, String.format(
           "Missing top state for partition %s beyond %s time. Graceful: %s",
           partitionName, missingDuration, false));
```

Comment:
partition level is too much. It will cause log flooding. If you would like to have it at info level, you can add the partition name in a map.  Print the map by end of the loop with one log. Log every partition could be slowing down the computation a lot.